services/motion/builtin/builtin.go

```diff
@@ -15,49 +16,76 @@ import (
 	"go.viam.com/rdk/operation"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/resource"
-	"go.viam.com/rdk/robot"
 	"go.viam.com/rdk/robot/framesystem"
 	"go.viam.com/rdk/services/motion"
 	"go.viam.com/rdk/spatialmath"
-	"go.viam.com/rdk/utils"
 )
 
 func init() {
 	resource.RegisterDefaultService(
 		motion.Subtype,
 		resource.DefaultServiceModel,
-		resource.Registration[motion.Service, resource.NoNativeConfig]{
-			DeprecatedRobotConstructor: func(
+		resource.Registration[motion.Service, *Config]{
+			Constructor: func(
 				ctx context.Context,
-				r any,
+				deps resource.Dependencies,
 				conf resource.Config,
 				logger golog.Logger,
 			) (motion.Service, error) {
-				actualR, err := utils.AssertType[robot.Robot](r)
-				if err != nil {
-					return nil, err
-				}
-				return NewBuiltIn(ctx, actualR, conf, logger)
+				return NewBuiltIn(ctx, deps, conf, logger)
 			},
 		})
 }
 
+// Config describes how to configure the service; currently only used for specifying dependency on framesystem service
+type Config struct {
+}
+
+// components will be depended upon weakly due to the above matcher.
+func (c *Config) Validate(path string) ([]string, error) {
```

Comment:
[nit] Can this comment on this function indicate that it returning the dependencies?